File: 2-string/valid-anagram.py
"""
Leet code 242: Valid Anagram
Given two strings s and t, return true if t is an anagram of s.
Techniques: Counter
Time complexity: O(n)
Space complexity: O(1)
"""


class Solution(object):
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        counter_s = [0] * 26
        counter_t = [0] * 26
        for i in range(len(s)):
            counter_s[ord(s[i]) - ord("a")] += 1
            counter_t[ord(t[i]) - ord("a")] += 1
        return counter_s == counter_t

File: 2-string/valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length !== t.length) return false
    counter_s = new Array(26).fill(0)
    counter_t = new Array(26).fill(0)
    const start = 'a'.charCodeAt(0)
    for (let i=0; i<s.length; i++) {
        counter_s[s.charCodeAt(i) - start] += 1
        counter_t[t.charCodeAt(i) - start] += 1
    }
    console.log(counter_s)
    console.log(counter_t)
    return counter_s.toString() === counter_t.toString()
};